### ISSUE TITLE
Implements Spotify local API

### ISSUE THREAD

**Issue by jwallet** [jwallet/spy-spotify#22]
[local url](http://127.0.0.1:4381/remote/status.json?ref=&cors=&_=1514595871&oauth=NAowChgKB1Nwb3RpZnkSABoGmAEByAEBJedVR1oSFCFZu8y0-qz23flhUsWrKnKSIcM8&csrf=6d2fb75ffe7241e18660d7534cbe992a)

instead of using the process to get infos, use the local api to get these :
- [x] Get track title
- [x] Get artist name
- [x] Get album title
- [x] Get song length
- [x] Get current track position
- [x] Check if the track is an ad
- [x] Check if Spotify is playing/on pause
- [x] Get album art

with the track, artist and album, we can find other infos using last api if available :
- [x] get first genre tag (search for toptags/tag/name)
- [x] get album arts from them since spotify only provides 1 art instead of usual 4
   - [x] small (search for image[size='small])
   - [x] medium (search for image[size='medium'])
   - [x] large (search for image[size='large'])
   - [x] extralarge (search for image[size='extralarge'])
- [x] get track number on the album (search for tracks/track/name, get tracks/track/@rank)

**Comment by Imparium** [comment]
If you could use this to put labels in the music would be awesome. We could easily select musics by album.

**Comment by eleasarchriso** [comment]
Hi,

wanted to thank you for this great tool! I spent a little bit of time to integrate the Spotify local API (get currently playing track) and also the Spotify web API (get the metadata and images). It is not optimized, I had to upgrade to .NET 4.5 and did not invest much time into refactorings and such. Also it likely requires more testing as I only did test it to record full albums and not any playlist or similar.

But maybe it still helps: https://github.com/eleasarchriso/spy-spotify
